implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -162,15 +161,18 @@ private String findSubject(JWTAuthContextInfo authContextInfo, JwtClaims claimsS
             Object claimValue = findClaimValue(authContextInfo.getGroupsPath(), claimsSet.getClaimsMap(), pathSegments, 0);
 
             if (claimValue instanceof List) {
-                @SuppressWarnings("unchecked")
-                List<String> groups = List.class.cast(claimValue);
-                return groups;
-            } else if (pathSegments.length == 1 && SCOPE_CLAIM.equals(pathSegments[0])) {
-                if (claimValue instanceof String) {
-                    return Arrays.asList(((String) claimValue).split(" "));
+                try {
+                    @SuppressWarnings("unchecked")
+                    List<String> groups = List.class.cast(claimValue);
```

Comment:
This doesn't make sense. We do `if (claimValue instanceof List) { ... }`, so inside that `if`, we can safely do `List<String> groups = (List<String>) claimValue;`

Reply:
Well, not _entirely_ safely, because erasure, but you get the idea :-)

Reply:
@Ladicek, it is not guaranteed to be a String though, or may be I'm missing something :-) ?

Reply:
`List.class.cast(claimValue);` doesn't guarantee that either :-)

Reply:
I mean, what I'm trying to say is: `List.class.cast(claimValue);` is IMHO not idiomatic, and we should use `(List<String>) claimValue`.

Reply:
@Ladicek I'm so slow, finally got it :-), let me tweak it

Reply:
@Ladicek I thought you were concerned about the whole exception catch thing :-)

Reply:
@Ladicek done

Reply:
Ah, yea, catching the exception doesn't make sense either :-) The only way to verify that the list really contains strings and nothing but strings is to manually verify each element.

Plus, `List<String> groups = (List<String>)List.class.cast(claimValue);` is worse then the previous way, you're casting twice :-)

Reply:
I probably shouldn't have started complaining about this, because it was already there :-)

Reply:
@Ladicek  That was me doing it in a rash, forgot to drop the 1st cast.
Actually, just noticed it was changed [here](https://github.com/smallrye/smallrye-jwt/commit/6a9ed997232319c043bbb3ccb7adfe2d6dde5af6#diff-5f4020e440d6d2dd7b7ff27fbe83dd25), originally it was done with a plain List Of String cast, so I believe Sonar must've been complaining about it. Anyway, I'm happy just to use the original List Of String cast if you think it is a better pattern.
But yeah, the cast exception, I was hoping the compiler would be somehow smart enough to translate that into the individual checks I guess. Have to find a clean way to get this exception thrown, hate to iterate manually :-). Have something in mind ? I guess we just need to create a new List and feed the original List into it somehow

Reply:
I'm totally not a fan of what we're doing with Sonar. Anyway, please feel free to remove the last commit, restore `List.class.cast(...)` and let's be done with it :-)